<!doctype html><html><head><title>Projects</title><link rel=stylesheet type=text/css href=/bootstrap/css/bootstrap.min.css><link rel=stylesheet type=text/css href=/bootstrap/icons/bootstrap-icons.css><link rel=stylesheet type=text/css href=https://scverse.org/main.min.css><link rel=apple-touch-icon sizes=180x180 href=../img/favicon/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=../img/favicon/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=../img/favicon/favicon-16x16.png><link rel=manifest href=../img/favicon/site.webmanifest><link rel=mask-icon href=../img/favicon/safari-pinned-tab.svg color=#000000><link rel="shortcut icon" href=../img/favicon/favicon.ico><meta name=msapplication-TileColor content="#00aba9"><meta name=msapplication-config content="../img/favicon/browserconfig.xml"><meta name=theme-color content="#ffffff"><meta charset=utf-8><meta content="utf-8" http-equiv=encoding><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><script src=/bootstrap/js/bootstrap.bundle.min.js async></script>
<script data-goatcounter=https://scverse.goatcounter.com/count async src=//gc.zgo.at/count.js></script></head><body><header><div id=banner><div id=banner-content>ðŸš§ Under construction ðŸš§</div></div><div class=container style=max-width:850px><nav class="navbar navbar-expand-lg navbar-light"><div class=container-fluid><a class="navbar-brand row align-items-center m-0" href=/><div id=scverse-logo class=col></div><span class="logo-name col m-0 px-3">scverse</span></a>
<button class=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#navbarSupportedContent aria-controls=navbarSupportedContent aria-expanded=false aria-label="Toggle navigation">
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse justify-content-end" id=navbarSupportedContent><ul class="navbar-nav mb-2 mb-lg-0 nav-buttons"><li class=nav-item><a class="nav-link
active" href=/projects>Projects</a></li><li class=nav-item><a class=nav-link href=/learn>Learn</a></li><li class=nav-item><a class=nav-link href=/people>People</a></li><li class=nav-item><a class=nav-link href=/blog>Blog</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/about id=navbarDropdown role=button aria-expanded=false>About</a><ul class=dropdown-menu aria-labelledby=navbarDropdown><h3></h3><li><a class=dropdown-item href=/about>About scverse</a></li><li><a class=dropdown-item href=/about/mission>Mission statement</a></li><li><a class=dropdown-item href=/about/roles>Roles</a></li><li><a class=dropdown-item href=/about/code_of_conduct>Code of Conduct</a></li></ul></li><li class=nav-item><a id=join-button class="nav-link nav-button-hl" href=/join>Join</a></li></ul></div></div></nav></div></header><div id=wrapper><div id=content><div id=page-content><h1>Projects</h1><article class=post><div class=post-content id=projects-content><h2>Core projects</h2><div id=projects-list><div class=project-section><div class=project-img><img src=../img/libs/anndata_schema.svg></div><div class=project-info><div class=project-tile id=anndata-tile><div class=project-text><span class=project-name>anndata</span>
<span class=project-details>Anndata is a Python package for handling annotated data matrices in memory and on disk, positioned between pandas and xarray. anndata offers a broad range of computationally efficient features including, among others, sparse data support, lazy operations, and a PyTorch interface.</span></div></div><div class=project-links><a href=https://github.com/theislab/anndata target=_blank>GitHub</a>
<a href=https://anndata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/anndata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/anndata target=_blank>Conda</a></div></div></div><div class=project-section><div class=project-img><img src=../img/libs/mudata_flat.svg></div><div class=project-info><div class=project-tile id=mudata-tile><div class=project-text><span class=project-name>mudata</span>
<span class=project-details>MuData is a format for annotated multimodal datasets where each modality is represented by an AnnData object. MuData's reference implementation is in Python, and the cross-language functionality is achieved via HDF5-based .h5mu files with libraries in R and Julia.</span></div></div><div class=project-links><a href=https://github.com/scverse/mudata target=_blank>GitHub</a>
<a href=https://mudata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/mudata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/mudata target=_blank>Conda</a>
<a href=https://github.com/scverse/Muon.jl target=_blank>Muon.jl</a></div></div></div><div class=project-section><div class=project-img><img src=../img/icons/scanpy.svg></div><div class=project-info><div class=project-tile id=scanpy-tile><div class=project-text><span class=project-name>scanpy</span>
<span class=project-details>Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with anndata. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</span></div></div><div class=project-links><a href=https://github.com/theislab/scanpy target=_blank>GitHub</a>
<a href=https://scanpy.readthedocs.io/en/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scanpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/scanpy target=_blank>Conda</a></div></div></div><div class=project-section><div class=project-img><img src=../img/icons/muon.svg></div><div class=project-info><div class=project-tile id=muon-tile><div class=project-text><span class=project-name>muon</span>
<span class=project-details>muon is a Python framework for multimodal omics analysis. While there are many features that muon brings to the table, there are three key areas that its functionality is focused on.</span></div></div><div class=project-links><a href=https://github.com/scverse/muon target=_blank>GitHub</a>
<a href=https://muon.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://muon-tutorials.readthedocs.io/en/latest/ target=_blank>Tutorials</a>
<a href=https://pypi.org/project/muon/ target=_blank>PyPI</a>
<a href=https://gtca.github.io/muon/ target=_blank>Website</a></div></div></div><div class=project-section><div class=project-img><img src=../img/libs/scvi_tools_graph_model.svg></div><div class=project-info><div class=project-tile id=scvi-tools-tile><div class=project-text><span class=project-name>scvi-tools</span>
<span class=project-details>scvi-tools is a library for developing and deploying machine learning models based on PyTorch and AnnData. With an emphasis on probablistic models, scvi-tools steamlines the development process via training, data management, and user interface abstractions. scvi-tools also contains easy-to-use implementations of more than 14 state-of-the-art probabilistic models in the field.</span></div></div><div class=project-links><a href=https://github.com/yoseflab/scvi-tools target=_blank>GitHub</a>
<a href=https://docs.scvi-tools.org/en/stable/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scvi-tools/ target=_blank>PyPI</a>
<a href=https://scvi-tools.org/ target=_blank>Website</a></div></div></div><div class=project-section><div class=project-img><img src=../img/icons/scirpy.svg></div><div class=project-info><div class=project-tile id=scirpy-tile><div class=project-text><span class=project-name>scirpy</span>
<span class=project-details>Scirpy is a scalable toolkit to analyse T-cell receptor or B-cell receptor repertoires from single-cell RNA sequencing data. It seamlessly integrates with scanpy and provides various modules for data import, analysis and visualization.</span></div></div><div class=project-links><a href=https://github.com/icbi-lab/scirpy target=_blank>GitHub</a>
<a href=https://icbi-lab.github.io/scirpy/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scirpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/bioconda/scirpy target=_blank>Conda</a></div></div></div><div class=project-section><div class=project-img><img src=../img/icons/squidpy.svg></div><div class=project-info><div class=project-tile id=squidpy-tile><div class=project-text><span class=project-name>squidpy</span>
<span class=project-details>Squidpy is a tool for the analysis and visualization of spatial molecular data. It builds on top of scanpy and anndata, from which it inherits modularity and scalability. It provides analysis tools that leverages the spatial coordinates of the data, as well as tissue images if available.</span></div></div><div class=project-links><a href=https://squidpy.readthedocs.io/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/squidpy/ target=_blank>PyPI</a></div></div></div></div><h2>Ecosystem</h2><div><p>Many popular packages rely on scverse functionality. For instance, they take advantage of established data format standards such as anndata and mudata, or are designed to be integrated into the workflow of analysis frameworks. Here are just a few of them.</p></div><div class=project-links><a href=https://github.com/Teichlab/bbknn target=_blank>bbknn</a>
<a href=https://cell2location.readthedocs.io target=_blank>cell2location</a>
<a href=https://github.com/broadinstitute/CellBender target=_blank>CellBender</a>
<a href=https://morris-lab.github.io/CellOracle.documentation/ target=_blank>CellOracle</a>
<a href=https://github.com/Teichlab/cellphonedb target=_blank>CellPhoneDB</a>
<a href=https://github.com/theislab/cellrank target=_blank>CellRank</a>
<a href=https://github.com/chanzuckerberg/cellxgene target=_blank>cellxgene</a>
<a href=https://github.com/interactivereport/cellxgene_VIP target=_blank>cellxgene_VIP</a>
<a href=https://github.com/gao-lab/Cell_BLAST target=_blank>Cell_BLAST</a>
<a href=https://cirrocumulus.readthedocs.io/en/latest/ target=_blank>Cirrocumulus</a>
<a href=https://github.com/dylkot/cNMF target=_blank>cNMF</a>
<a href=https://github.com/facebookresearch/CPA target=_blank>CPA</a>
<a href=https://github.com/theislab/dca target=_blank>dca</a>
<a href=https://github.com/saezlab/decoupler-py target=_blank>decoupler</a>
<a href=https://github.com/theislab/diffxpy target=_blank>diffxpy</a>
<a href=https://github.com/JonathanShor/DoubletDetection target=_blank>DoubletDetection</a>
<a href=https://github.com/aristoteleo/dynamo-release target=_blank>dynamo-release</a>
<a href=https://github.com/colomemaria/epiScanpy target=_blank>epiScanpy</a>
<a href=http://cb.csail.mit.edu/cb/scanorama/ target=_blank>scanorama</a>
<a href=https://github.com/kaizhang/SnapATAC2 target=_blank>SnapATAC2</a>
<a href=https://github.com/snap-stanford/mars target=_blank>MARS</a>
<a href=https://novosparc.readthedocs.io target=_blank>novoSpaRc</a>
<a href=https://openproblems.bio target=_blank>Open Problems</a>
<a href=https://pathml.org target=_blank>PathML</a>
<a href=https://github.com/dpeerlab/Palantir target=_blank>Palantir</a>
<a href=https://github.com/raphael-group/paste target=_blank>PASTE</a>
<a href=https://pegasus.readthedocs.io/en/stable/ target=_blank>pegasus</a>
<a href=https://github.com/aertslab/pySCENIC target=_blank>pySCENIC</a>
<a href=https://sccoda.readthedocs.io/en/latest/ target=_blank>scCODA</a>
<a href=https://github.com/theislab/scarches target=_blank>scArches</a>
<a href=https://github.com/theislab/scgen target=_blank>scGen</a>
<a href=https://github.com/swolock/scrublet target=_blank>scrublet</a>
<a href=https://github.com/theislab/scvelo target=_blank>scvelo</a>
<a href=https://github.com/theislab/sfaira target=_blank>sfaira</a>
<a href=https://github.com/jianhuupenn/SpaGCN target=_blank>SpaGCN</a>
<a href=https://github.com/theislab/squidpy target=_blank>squidpy</a>
<a href=https://tangram-sc.readthedocs.io/en/latest/index.html target=_blank>Tangram</a>
<a href=https://github.com/vitessce/vitessce target=_blank>vitesse</a>
<a href=https://github.com/broadinstitute/wot%29 target=_blank>wot</a></div></div></article></div></div></div><footer><div id=footer-content class=container><div class=row><div class="col-12 col-md-6 col-lg-3"><ul><h5>Pages</h5><li><a href=/projects>Projects</a></li><li><a href=/learn>Learn</a></li><li><a href=/people>People</a></li><li><a href=/blog>Blog</a></li><li><a href=/about>About</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Governance</h5><li><a href=/about/mission/>Mission statement</a></li><li><a href=/about/code_of_conduct>Code of conduct</a></li><li><a href=/about/roles>Roles</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Join scverse</h5><li><a href=https://github.com/scverse target=_blank>GitHub</a></li><li><a href=https://discourse.scverse.org/ target=_blank>Discourse</a></li><li><a href=https://scverse.zulipchat.com/ target=_blank>Zulip</a></li><li><a href=https://twitter.com/scanpy_team target=_blank>Twitter</a></li><li><a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA target=_blank>YouTube</a></li></ul></div></div><div class=signature><p>scverse core team, 2022</p></div></div></footer></body></html>