<!doctype html><html><head><title>Getting Started</title><link rel=stylesheet type=text/css href=/bootstrap/css/bootstrap.min.css><link rel=stylesheet type=text/css href=/bootstrap/icons/bootstrap-icons.css><link rel=stylesheet type=text/css href=https://scverse.org/main.min.css><link rel=apple-touch-icon sizes=180x180 href=../img/favicon/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=../img/favicon/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=../img/favicon/favicon-16x16.png><link rel=manifest href=../img/favicon/site.webmanifest><link rel=mask-icon href=../img/favicon/safari-pinned-tab.svg color=#000000><link rel="shortcut icon" href=../img/favicon/favicon.ico><meta name=msapplication-TileColor content="#00aba9"><meta name=msapplication-config content="../img/favicon/browserconfig.xml"><meta name=theme-color content="#ffffff"><meta charset=utf-8><meta content="utf-8" http-equiv=encoding><meta name=twitter:card content="summary"><meta name=twitter:title content="Getting Started"><meta name=twitter:description content><meta property="og:title" content="Getting Started"><meta property="og:description" content><meta property="og:type" content="website"><meta property="og:url" content="https://scverse.org/learn/"><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><script src=/main.js async></script>
<script src=/bootstrap/js/bootstrap.bundle.min.js async></script>
<script data-goatcounter=https://scverse.goatcounter.com/count async src=//gc.zgo.at/count.js></script></head><body><header><div class=container style=max-width:850px><nav class="navbar navbar-expand-lg navbar-light"><div class=container-fluid><a class="navbar-brand row align-items-center m-0" href=/><div id=scverse-logo class=col></div></a><button class=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#navbarSupportedContent aria-controls=navbarSupportedContent aria-expanded=false aria-label="Toggle navigation">
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse justify-content-end" id=navbarSupportedContent><ul class="navbar-nav mb-2 mb-lg-0 nav-buttons"><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/packages id=packages-dropdown role=button aria-expanded=false>Packages</a><ul class=dropdown-menu aria-labelledby=packages-dropdown><li><a class=dropdown-item href=/packages#core-packages>Core</a></li><li><a class=dropdown-item href=/packages/#ecosystem>Ecosystem</a></li></ul></li><li class=nav-item><a class="nav-link
active" href=/learn>Learn</a></li><li class=nav-item><a class=nav-link href=/people>People</a></li><li class=nav-item><a class=nav-link href=/blog>Blog</a></li><li class=nav-item><a class=nav-link href=/events>Events</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/about id=about-dropdown role=button aria-expanded=false>About</a><ul class=dropdown-menu aria-labelledby=about-dropdown><li><a class=dropdown-item href=/about>About scverse</a></li><li><a class=dropdown-item href=/about/mission>Mission statement</a></li><li><a class=dropdown-item href=/about/roles>Roles</a></li><li><a class=dropdown-item href=/about/code_of_conduct>Code of Conduct</a></li></ul></li><li class=nav-item><a id=join-button class="nav-link nav-button-hl" href=/join>Join</a></li></ul></div></div></nav></div></header><div id=wrapper><div id=content><div id=page-content><h1>Getting Started</h1><article class=post><div class=post-content id=tutorials-content><p>If you are new to the scverse, get started with this set of tutorials covering basic analysis and functionality of the core pacakges.
For more tutorials as well as API documentation and user guides, see the sites of <a href=/packages/>individual packages</a>.</p><p>You can also find recordings of past talks and workshops on our <a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA>YouTube channel</a>.</p><h2 id=tutorials>Tutorials</h2><div id=ecosystem-tutorials><input type=text class=form-control id=tutorial-filter onkeyup=filterTutorials() placeholder="Search through 13 tutorials" title="Type in your search query"><h3>Data structures</h3><p>These tutorials teach you how to work with scverse data structures.
If you are new to Python and/or scverse, we recommend you read the
"getting started" and "axes" tutorials first.</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/scverse_data_backed.html#working-with-scverse-objects-in-backed-mode target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/scverse-objects-in-backed-mode/icon.png><div class=card-body><h5 class=card-title>Working with scverse objects in backed mode</h5><p>In this tutorial we show how to work with scverse data objects without loading the full
datasets in memory. AnnData and MuData objects are saved to disk as .h5ad and .h5mu files.
These formats save HDF5 stores (via h5py) that can be accessed without loading heavy data
matrices in memory (i.e. in backed mode), as a “read-only” object.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/tutorial_axes_anndata_mudata.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/anndata-mudata-axes/icon.svg><div class=card-body><h5 class=card-title>Axes in AnnData and MuData</h5><p>In this tutorial we will showcase operations on AnnData objects storing each one independent view
of the same scRNAseq matrix (and accompanying metadata) and show how different choices of processing
workflows can be stored in the same MuData object.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/anndata_getting_started.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/anndata-getting-started/icon.svg><div class=card-body><h5 class=card-title>Getting started with AnnData</h5><p>Suppose a colleague of yours did some single cell data analysis in Python and Scanpy,
saving the results in an AnnData object and sending it to you in a *.h5ad file. This tutorial will walk you
through that file and help you explore its structure and content — even if you are new to anndata, Scanpy or Python.</p></div></a></div></div><h3>scRNA-seq</h3><p>The following tutorials show show to analyze single-cell gene expression data.</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://anndata.readthedocs.io/en/latest/concatenation.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/concatenation-of-unimodal-data/icon.png><div class=card-body><h5 class=card-title>Concatenation</h5><p>In this notebook we showcase how to perform concatenation,
meaning to keep all sub elements of each object, and stack these elements in an ordered way.
Merging is combining a set of collections into one resulting collection which contains
elements from the objects.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://muon-tutorials.readthedocs.io/en/latest/cite-seq/index.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/CITEseq-integration/icon.png><div class=card-body><h5 class=card-title>CITE-seq integration</h5><p>CITE-seq is a method for cellular indexing of transcriptomes and
epitopes by sequencing. CITE-seq datasets comprise transcriptome-wide
measurements for single cells (gene expression) as well as surface
protein level information, typically for a few dozens of proteins.
These notebooks cover different CITE-seq datasets:
- analysis of 5k peripheral blood mononuclear cells (PBMCs) showcasing
dsb normalisation and data integration with MOFA+
- calculating weighted nearest neighbours for the 5k PBMCs dataset
that also demonstrates how to handle multiple multimodal embeddings.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://decoupler-py.readthedocs.io/en/latest/notebooks/pseudobulk.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/decoupler-pseudobulk-de/icon.png><div class=card-body><h5 class=card-title>Pseudo-bulk differential expression and functional analysis</h5><p>In this notebook we showcase how to use decoupler for pathway and transcription factor (TF) enrichment from
a human data-set. The data consists of ~5k Blood myeloid cells from healthy and COVID-19 infected patients
available in the Single Cell Expression Atlas here.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scvelo.readthedocs.io/en/stable/getting_started/ target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/RNA-velocity/icon.png><div class=card-body><h5 class=card-title>RNA velocity</h5><p>This tutorial guides you through how RNA velocity can be inferred from single cell RNA-seq data
using scVelo.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/trajectories/pseudotemporal.html# target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/pseudotemporal-ordering/icon.png><div class=card-body><h5 class=card-title>Pseudotemporal ordering</h5><p>This tutorial show how a pseudotime can be constructed and compares different pseudotimes.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/conditions/compositional.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/compositional-analysis/icon.png><div class=card-body><h5 class=card-title>Compositional analysis</h5><p>Compositional analysis can be done on the level of cell identity clusters
in the form of known cell types or cell states corresponding to, for example, cells recently affected by perturbations.
This chapter will introduce both approaches and apply them to the Haber dataset.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/basic-scrna-tutorial.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/preprocessing-and-clustering/icon.webp><div class=card-body><h5 class=card-title>Preprocessing, clustering and cell-type annotation</h5><p>This fundamental single-cell tutorial guides you through the most common analysis steps, including quality control,
normalization, feature selection, dimensionality reduction, clustering and cell-type annotation.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://docs.scvi-tools.org/en/stable/tutorials/notebooks/api_overview.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/scvi-batch-effect-removal/icon.svg><div class=card-body><h5 class=card-title>Batch-effect removal with scvi-tools</h5><p>In this tutorial, we demonstrate how to use scvi-tools to fit a model to single-cell count data,
correct batch effects, and perform differential gene expression analysis.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/tutorial_concatenation_anndata_mudata.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/concatenation-of-multimodal-data/icon.png><div class=card-body><h5 class=card-title>Concatenation of multimodal data</h5><p>This tutorial shows how you can concatenate 2 MuData objects that may represent complementary
slices of the same dataset or 2 modalities into one AnnData.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/conditions/perturbation_modeling.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/perturbation-modeling/icon.png><div class=card-body><h5 class=card-title>Perturbation modeling</h5><p>This tutorial introduces three approaches that can be tackled with
single-cell perturbation data in the following sections:
- Finding the cell types that were most affected by pertubations using Augur
- Predicting the transcriptional response of single cells to perturbations using scGen
- Quantifying the sensitivity of genetic CRISPR perturbations using Mixscape</p></div></a></div></div></div></div></article></div></div></div><footer><div id=footer-content class=container><div class=row><div class="col-12 col-md-6 col-lg-3"><ul><h5>Pages</h5><li><a href=/packages>Packages</a></li><li><a href=/learn>Learn</a></li><li><a href=/people>People</a></li><li><a href=/blog>Blog</a></li><li><a href=/events>Events</a></li><li><a href=/about>About</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Governance</h5><li><a href=/about/mission/>Mission statement</a></li><li><a href=/about/code_of_conduct>Code of conduct</a></li><li><a href=/about/roles>Roles</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Join scverse</h5><li><a href=https://github.com/scverse target=_blank>GitHub</a></li><li><a href=https://discourse.scverse.org/ target=_blank>Discourse</a></li><li><a href=https://scverse.zulipchat.com/ target=_blank>Zulip</a></li><li><a href=https://twitter.com/scverse_team target=_blank>Twitter</a></li><li><a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA target=_blank>YouTube</a></li></ul></div></div><div class=signature><p>scverse core team, 2023</p></div></div></footer></body></html>