<!doctype html><html><head><title>Getting Started</title>
<link rel=stylesheet type=text/css href=/bootstrap/css/bootstrap.min.css><link rel=stylesheet type=text/css href=/bootstrap/icons/bootstrap-icons.css><link rel=stylesheet type=text/css href=https://scverse.org/main.min.css><link rel=apple-touch-icon sizes=180x180 href=../img/favicon/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=../img/favicon/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=../img/favicon/favicon-16x16.png><link rel=manifest href=../img/favicon/site.webmanifest><link rel=mask-icon href=../img/favicon/safari-pinned-tab.svg color=#000000><link rel="shortcut icon" href=../img/favicon/favicon.ico><meta name=msapplication-TileColor content="#00aba9"><meta name=msapplication-config content="../img/favicon/browserconfig.xml"><meta name=theme-color content="#ffffff"><meta charset=utf-8><meta content="utf-8" http-equiv=encoding><meta name=twitter:card content="summary"><meta name=twitter:title content="Getting Started"><meta property="og:url" content="https://scverse.org/learn/"><meta property="og:site_name" content="scverse"><meta property="og:title" content="Getting Started"><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><script src=/main.js async></script><script src=/bootstrap/js/bootstrap.bundle.min.js async></script><script data-goatcounter=https://scverse.goatcounter.com/count async src=//gc.zgo.at/count.js></script></head><body><header><div class=container style=max-width:850px><nav class="navbar navbar-expand-lg navbar-light"><div class=container-fluid><a class="navbar-brand row align-items-center m-0" href=/><div id=scverse-logo class=col></div></a><button class=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#navbarSupportedContent aria-controls=navbarSupportedContent aria-expanded=false aria-label="Toggle navigation">
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse justify-content-end" id=navbarSupportedContent><ul class="navbar-nav mb-2 mb-lg-0 nav-buttons"><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/packages id=packages-dropdown role=button aria-expanded=false>Packages</a><ul class=dropdown-menu aria-labelledby=packages-dropdown><li><a class=dropdown-item href=/packages#core-packages>Core</a></li><li><a class=dropdown-item href=/packages/#ecosystem>Ecosystem</a></li></ul></li><li class=nav-item><a class="nav-link
active" href=/learn>Learn</a></li><li class=nav-item><a class=nav-link href=/people>People</a></li><li class=nav-item><a class=nav-link href=/blog>Blog</a></li><li class=nav-item><a class=nav-link href=/events>Events</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/about id=about-dropdown role=button aria-expanded=false>About</a><ul class=dropdown-menu aria-labelledby=about-dropdown><li><a class=dropdown-item href=/about>About scverse</a></li><li><a class=dropdown-item href=/about/mission>Mission statement</a></li><li><a class=dropdown-item href=/about/roles>Roles</a></li><li><a class=dropdown-item href=/about/code_of_conduct>Code of Conduct</a></li></ul></li><li class=nav-item><a id=join-button class="nav-link nav-button-hl" href=/join>Join</a></li></ul></div></div></nav></div></header><div id=wrapper><div id=content><div id=page-content><h1>Getting Started</h1><article class=post><div class=post-content id=tutorials-content><p>If you are new to the scverse, get started with this set of tutorials covering basic analysis and functionality of the core pacakges.
For more tutorials as well as API documentation and user guides, see the sites of <a href=/packages/>individual packages</a>.</p><p>You can also find recordings of past talks and workshops on our <a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA>YouTube channel</a>.</p><h2 id=tutorials>Tutorials</h2><div id=ecosystem-tutorials><input type=text class=form-control id=tutorial-filter onkeyup=filterTutorials() placeholder="Search through 22 tutorials" title="Type in your search query"><h3>Data structures</h3><p>These tutorials teach you how to work with scverse data structures.
If you are new to Python and/or scverse, we recommend you read the
"getting started" and "axes" tutorials first.</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/scverse_data_backed.html#working-with-scverse-objects-in-backed-mode target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/scverse-objects-in-backed-mode/icon.png><div class=card-body><h5 class=card-title>Working with scverse objects in backed mode</h5><p>In this tutorial, we demonstrate working with scverse data objects
without loading full datasets. (AnnData and MuData are saved as .h5ad and .h5mu files)</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/tutorial_concatenation_anndata_mudata.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/concatenation-of-multimodal-data/icon.png><div class=card-body><h5 class=card-title>Concatenation of multimodal data</h5><p>This tutorial shows how you can concatenate 2 MuData objects that may represent complementary
slices of the same dataset or 2 modalities into one AnnData.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/anndata_getting_started.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/anndata-getting-started/icon.svg><div class=card-body><h5 class=card-title>Getting started with AnnData</h5><p>This tutorial helps you to explore the structure and content of single-cell
data analysis results in a *.h5ad file using AnnData, Scanpy, and Python.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://anndata.readthedocs.io/en/latest/concatenation.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/concatenation-of-unimodal-data/icon.png><div class=card-body><h5 class=card-title>Concatenation</h5><p>In this notebook we showcase how to perform concatenation, meaning to
keep all sub elements of each object, and stack these elements in an
ordered way.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/tutorial_axes_anndata_mudata.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/anndata-mudata-axes/icon.svg><div class=card-body><h5 class=card-title>Axes in AnnData and MuData</h5><p>In this tutorial we showcase operations on independent AnnData objects
(scRNAseq matrix + metadata), demonstrating how various processing
workflows can be stored in one MuData object.</p></div></a></div></div><h3>scRNA-seq</h3><p>The following tutorials show show to analyze single-cell gene expression data.</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://scvelo.readthedocs.io/en/stable/getting_started.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/RNA-velocity/icon.png><div class=card-body><h5 class=card-title>RNA velocity</h5><p>This tutorial guides you through how RNA velocity can be inferred from single cell RNA-seq data
using scVelo.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/basic-scrna-tutorial.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/preprocessing-and-clustering/icon.webp><div class=card-body><h5 class=card-title>Preprocessing, clustering and cell-type annotation</h5><p>This fundamental tutorial covers common analysis steps: quality control,
normalization, feature selection, dimensionality reduction, clustering,
and cell-type annotation.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/conditions/compositional.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/compositional-analysis/icon.png><div class=card-body><h5 class=card-title>Compositional analysis</h5><p>This tutorial introduces compositional analysis at cell identity
cluster level, based on known cell types or states affected by
perturbations.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/trajectories/pseudotemporal.html# target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/pseudotemporal-ordering/icon.png><div class=card-body><h5 class=card-title>Pseudotemporal ordering</h5><p>This tutorial show how a pseudotime can be constructed and compares different pseudotimes.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://docs.scvi-tools.org/en/stable/tutorials/notebooks/quick_start/api_overview.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/scvi-batch-effect-removal/icon.svg><div class=card-body><h5 class=card-title>Batch-effect removal with scvi-tools</h5><p>In this tutorial, we demonstrate how to use scvi-tools to fit a model to single-cell count data,
correct batch effects, and perform differential gene expression analysis.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://decoupler-py.readthedocs.io/en/latest/notebooks/pseudobulk.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/decoupler-pseudobulk-de/icon.png><div class=card-body><h5 class=card-title>Pseudo-bulk differential expression and functional analysis</h5><p>This notebook showcases decoupler for pathway and TF enrichment on ~5k
Blood myeloid cells from healthy and COVID-19 infected patients.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://www.sc-best-practices.org/conditions/perturbation_modeling.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/perturbation-modeling/icon.png><div class=card-body><h5 class=card-title>Perturbation modeling</h5><p>This tutorial covers 3 approaches using single-cell perturbation data:
Augur (identify affected cell types), scGen (predict transcriptional response),
Mixscape (quantify CRISPR sensitivity).</p></div></a></div></div><h3>Spatial</h3><p>Analyze spatial data generated with different technologies</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://bento-tools.readthedocs.io/en/latest/index.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/squidpy-spatial/icon.png><div class=card-body><h5 class=card-title>Spatial analysis at subcellular resolution</h5><p>This tutorial shows how to use bentotools to study
gene expression at subcellular resolution.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://squidpy.readthedocs.io/en/latest/notebooks/tutorials/tutorial_vizgen_mouse_liver.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/bentotools-subcellular-resolution/icon.png><div class=card-body><h5 class=card-title>Spatial analysis with squidpy</h5><p>This tutorial demonstrate how to use squidpy to analyse transcriptomics
data with spatial resolution.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://cellcharter.readthedocs.io/en/latest/notebooks/cosmx_human_nsclc.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/cellcharter-spatial-transcriptomics/icon.png><div class=card-body><h5 class=card-title>Spatial clustering of spacial transcriptomics data with CellCharter</h5><p>This tutorial demonstrate how to use CellCharter to cluster spatial transcriptomics data, obtained with the CosMx technology from Nanostring.
This method can also be used for spatial proteomics data.</p></div></a></div></div><h3>Adaptive immune cell receptor</h3><p>Tutorials for analyzing single-cell B-cell and T-cell receptor sequencing data</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://scirpy.scverse.org/en/latest/tutorials/tutorial_3k_tcr.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/scirpy-tcr/icon.svg><div class=card-body><h5 class=card-title>Single-cell T-cell receptor analysis with scirpy</h5><p>In this tutorial, we show how to perfrom QC on scTCR-seq data,
define clonotype, cluster receptors by their sequence similarity
and compute repertoire overlaps between patients.</p></div></a></div></div><h3>Surface proteins</h3><p>CITE-seq analyses</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://muon-tutorials.readthedocs.io/en/latest/cite-seq/1-CITE-seq-PBMC-5k.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/CITEseq-integration/icon.png><div class=card-body><h5 class=card-title>CITE-seq integration</h5><p>These notebooks showcase CITE-seq analysis of PBMCs with dsb
normalization, MOFA+ data integration, and weighted nearest neighbors
handling multimodal embeddings.</p></div></a></div></div><h3>ATAC-seq</h3><p>Analyse chromatin accessibility data</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://muon-tutorials.readthedocs.io/en/latest/single-cell-rna-atac/pbmc10k/2-Chromatin-Accessibility-Processing.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/ATAC-preprocessing/icon.png><div class=card-body><h5 class=card-title>Processing chromatin accessibility</h5><p>This chapter shows multimodal single-cell gene expression and
chromatin accessibility analysis. In this notebook, scATAC-seq
data processing is described.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://docs.scvi-tools.org/en/stable/tutorials/notebooks/multimodal/MultiVI_tutorial.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/Joint-analysis-of-multiomic-data-with-MultiVI/icon.png><div class=card-body><h5 class=card-title>Joint analysis of paired and unpaired multiomic data with MultiVI</h5><p>This tutorial shows how to read multiomic data, create a joint object with
paired/unpaired data, train MultiVI model, visualize latent space,
and run differential analyses.</p></div></a></div></div><h3>Tips & Tricks</h3><p>Useful tips for data analysis with scverse tools that are independent of specific packages.</p><div class=tutorials-list><div class="card tutorial-item border-0"><a href=https://scverse-tutorials.readthedocs.io/en/latest/notebooks/scverse_data_interoperability.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/interoperability/icon.png><div class=card-body><h5 class=card-title>Interoperability</h5><p>This document lists resources for conversion to other data formats and programming
languages, e.g. R, Julia, ...</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scanpy-tutorials.readthedocs.io/en/latest/plotting/advanced.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/advanced-plotting/icon.png><div class=card-body><h5 class=card-title>Advanced plotting</h5><p>This tutorial explains how to customize matplotlib plots generated
by scanpy or other scverse libraries.</p></div></a></div><div class="card tutorial-item border-0"><a href=https://scanpy-tutorials.readthedocs.io/en/latest/plotting/core.html target=_blank><img class=card-img-top src=https://scverse.org/scverse-tutorials/plotting-in-scanpy/icon.png><div class=card-body><h5 class=card-title>Plotting in scanpy</h5><p>This tutorial explores the visualization possibilities of scanpy, including
embeddings and the visualization of marker genes and differentially expressed genes.</p></div></a></div></div></div></div></article></div></div></div><footer><div id=footer-content class=container><div class=row><div class="col-12 col-md-6 col-lg-3"><ul><h5>Pages</h5><li><a href=/packages>Packages</a></li><li><a href=/learn>Learn</a></li><li><a href=/people>People</a></li><li><a href=/blog>Blog</a></li><li><a href=/events>Events</a></li><li><a href=/about>About</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Governance</h5><li><a href=/about/mission/>Mission statement</a></li><li><a href=/about/code_of_conduct>Code of conduct</a></li><li><a href=/about/roles>Roles</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Join scverse</h5><li><a href=https://github.com/scverse target=_blank>GitHub</a></li><li><a href=https://discourse.scverse.org/ target=_blank>Discourse</a></li><li><a href=https://scverse.zulipchat.com/ target=_blank>Zulip</a></li><li><a href=https://x.com/scverse_team target=_blank>X (Twitter)</a></li><li><a href=https://bsky.app/profile/scverse.bsky.social target=_blank>Bluesky</a></li><li><a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA target=_blank>YouTube</a></li></ul></div></div><div class=signature><p>scverse core team, 2025</p></div></div></footer></body></html>