<!doctype html><html><head><title>Packages</title><link rel=stylesheet type=text/css href=/bootstrap/css/bootstrap.min.css><link rel=stylesheet type=text/css href=/bootstrap/icons/bootstrap-icons.css><link rel=stylesheet type=text/css href=/main.min.css><link rel=apple-touch-icon sizes=180x180 href=/img/favicon/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=/img/favicon/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=/img/favicon/favicon-16x16.png><link rel=manifest href=/img/favicon/site.webmanifest><link rel=mask-icon href=/img/favicon/safari-pinned-tab.svg color=#000000><link rel="shortcut icon" href=/img/favicon/favicon.ico><meta name=msapplication-TileColor content="#00aba9"><meta name=msapplication-config content="/img/favicon/browserconfig.xml"><meta name=theme-color content="#ffffff"><meta charset=utf-8><meta content="utf-8" http-equiv=encoding><meta name=twitter:card content="summary"><meta name=twitter:title content="Packages"><meta property="og:url" content="https://scverse.org/packages/"><meta property="og:site_name" content="scverse"><meta property="og:title" content="Packages"><meta property="og:locale" content="en_us"><meta property="og:type" content="website"><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><script src=/main.js async></script><script src=/bootstrap/js/bootstrap.bundle.min.js async></script><script data-goatcounter=https://scverse.goatcounter.com/count async src=//gc.zgo.at/count.js></script></head><body><header><div class=container style=max-width:850px><nav class="navbar navbar-expand-lg navbar-light"><div class=container-fluid><a class="navbar-brand row align-items-center m-0" href=/><div id=scverse-logo class=col></div><span class=visually-hidden>scverse</span>
</a><button class=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#navbarSupportedContent aria-controls=navbarSupportedContent aria-expanded=false aria-label="Toggle navigation">
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse justify-content-end" id=navbarSupportedContent><ul class="navbar-nav mb-2 mb-lg-0 nav-buttons"><li class="nav-item dropdown"><a class="nav-link dropdown-toggle
active" href=/packages id=packages-dropdown role=button aria-expanded=false>Packages</a><ul class=dropdown-menu aria-labelledby=packages-dropdown><li><a class=dropdown-item href=/packages#core-packages>Core</a></li><li><a class=dropdown-item href=/packages/#ecosystem>Ecosystem</a></li></ul></li><li class=nav-item><a class=nav-link href=/learn>Learn</a></li><li class=nav-item><a class=nav-link href=/people>People</a></li><li class=nav-item><a class=nav-link href=/blog>Blog</a></li><li class=nav-item><a class=nav-link href=/events>Events</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/about id=about-dropdown role=button aria-expanded=false>About</a><ul class=dropdown-menu aria-labelledby=about-dropdown><li><a class=dropdown-item href=/about>About scverse</a></li><li><a class=dropdown-item href=/about/mission>Mission statement</a></li><li><a class=dropdown-item href=/about/roles>Roles</a></li><li><a class=dropdown-item href=/about/code_of_conduct>Code of Conduct</a></li></ul></li><li class=nav-item><a id=join-button class="nav-link nav-button-hl" href=/join>Join</a></li></ul></div></div></nav></div></header><div id=wrapper><div id=content><div id=page-content><h1>Packages</h1><article class=post><div class=post-content id=packages-content><h2 id=core-packages>Packages maintained by core team</h2>These packages are considered foundational in that many other packages build upon them. Joint maintenance by the core team guarantees long-term stability.<h3 id=datastructures>Data structures</h3>Data structures are the foundational building block for all scverse packages. Building upon common data structures ensures interoperability.<div id=packages-list><div class=package-section><div class=package-img><img src=../img/libs/anndata_schema.svg alt="Logo for anndata"></div><div class=package-info><div class=package-tile id=anndata-tile><div class=package-text><span class=package-name>anndata</span>
<span class=package-details>AnnData is a Python package for handling annotated data matrices in memory and on disk, positioned between pandas and xarray. anndata offers a broad range of computationally efficient features including, among others, sparse data support, lazy operations, and a PyTorch interface.</span></div></div><div class=package-links><a href=https://github.com/theislab/anndata target=_blank>GitHub</a>
<a href=https://anndata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/anndata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/anndata target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/libs/mudata_flat.svg alt="Logo for mudata"></div><div class=package-info><div class=package-tile id=mudata-tile><div class=package-text><span class=package-name>mudata</span>
<span class=package-details>MuData is a format for annotated multimodal datasets where each modality is represented by an AnnData object. MuData&rsquo;s reference implementation is in Python, and the cross-language functionality is achieved via HDF5-based .h5mu files with libraries in R and Julia.</span></div></div><div class=package-links><a href=https://github.com/scverse/mudata target=_blank>GitHub</a>
<a href=https://mudata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/mudata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/mudata target=_blank>Conda</a>
<a href=https://github.com/scverse/Muon.jl target=_blank>Muon.jl</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/spatialdata.svg alt="Logo for spatialdata"></div><div class=package-info><div class=package-tile id=spatialdata-tile><div class=package-text><span class=package-name>spatialdata</span>
<span class=package-details>SpatialData is a data framework that comprises a FAIR storage format and a collection of python libraries for performant access, alignment, and processing of uni- and multi-modal spatial omics datasets. This repository contains the core spatialdata library. See the links below to learn more about other packages in the SpatialData ecosystem.</span></div></div><div class=package-links><a href=https://github.com/scverse/spatialdata target=_blank>GitHub</a>
<a href=https://spatialdata.scverse.org/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/spatialdata/ target=_blank>PyPI</a>
<a href=https://github.com/scverse/spatialdata-io target=_blank>spatialdata-io</a></div></div></div></div><h4>Analysis task-specific extensions</h4><p>In addition to these packages, we define standards on how to represent certain data types in these data structures. For now, such a specification is available for <a href=https://scirpy.scverse.org/en/latest/data-structure.html#storing-airr-rearrangement-data-in-anndata>Adaptive Immune Receptor Repertoire (AIRR) data</a>.</p><h3 id=frameworks>Frameworks</h3>Frameworks provide essential algorithms and plotting functions for specific analysis steps, building on our data structures.<div id=packages-list><div class=package-section><div class=package-img><img src=../img/icons/scanpy.svg alt="Logo for scanpy"></div><div class=package-info><div class=package-tile id=scanpy-tile><div class=package-text><span class=package-name>scanpy</span>
<span class=package-details>Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with anndata. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</span></div></div><div class=package-links><a href=https://github.com/theislab/scanpy target=_blank>GitHub</a>
<a href=https://scanpy.readthedocs.io/en/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scanpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/scanpy target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/muon.svg alt="Logo for muon"></div><div class=package-info><div class=package-tile id=muon-tile><div class=package-text><span class=package-name>muon</span>
<span class=package-details>muon is a Python framework for multimodal omics analysis. While there are many features that muon brings to the table, there are three key areas that its functionality is focused on.</span></div></div><div class=package-links><a href=https://github.com/scverse/muon target=_blank>GitHub</a>
<a href=https://muon.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://muon-tutorials.readthedocs.io/en/latest/ target=_blank>Tutorials</a>
<a href=https://pypi.org/project/muon/ target=_blank>PyPI</a>
<a href=https://muon.scverse.org/ target=_blank>Website</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/squidpy.svg alt="Logo for squidpy"></div><div class=package-info><div class=package-tile id=squidpy-tile><div class=package-text><span class=package-name>squidpy</span>
<span class=package-details>Squidpy is a tool for the analysis and visualization of spatial molecular data. It builds on top of scanpy and anndata, from which it inherits modularity and scalability. It provides analysis tools that leverages the spatial coordinates of the data, as well as tissue images if available.</span></div></div><div class=package-links><a href=https://github.com/theislab/squidpy target=_blank>GitHub</a>
<a href=https://squidpy.readthedocs.io/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/squidpy/ target=_blank>PyPI</a></div></div></div><div class=package-section><div class=package-img><img src=../img/libs/scvi_tools_graph_model.svg alt="Logo for scvi-tools"></div><div class=package-info><div class=package-tile id=scvi-tools-tile><div class=package-text><span class=package-name>scvi-tools</span>
<span class=package-details>scvi-tools is a library for developing and deploying machine learning models based on PyTorch and AnnData. With an emphasis on probablistic models, scvi-tools steamlines the development process via training, data management, and user interface abstractions. scvi-tools also contains easy-to-use implementations of more than 14 state-of-the-art probabilistic models in the field.</span></div></div><div class=package-links><a href=https://github.com/yoseflab/scvi-tools target=_blank>GitHub</a>
<a href=https://docs.scvi-tools.org/en/stable/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scvi-tools/ target=_blank>PyPI</a>
<a href=https://scvi-tools.org/ target=_blank>Website</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/scirpy.svg alt="Logo for scirpy"></div><div class=package-info><div class=package-tile id=scirpy-tile><div class=package-text><span class=package-name>scirpy</span>
<span class=package-details>Scirpy is a scalable toolkit to analyse T-cell receptor or B-cell receptor repertoires from single-cell RNA sequencing data. It seamlessly integrates with scanpy and provides various modules for data import, analysis and visualization.</span></div></div><div class=package-links><a href=https://github.com/icbi-lab/scirpy target=_blank>GitHub</a>
<a href=https://scirpy.scverse.org/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scirpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/bioconda/scirpy target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/snapatac2.svg alt="Logo for SnapATAC2"></div><div class=package-info><div class=package-tile id=SnapATAC2-tile><div class=package-text><span class=package-name>SnapATAC2</span>
<span class=package-details>SnapATAC2 is a scalable and modular pipeline for analyzing single-cell ATAC-seq data, enabling efficient preprocessing, dimensionality reduction, clustering, and integration with single-cell RNA-seq.</span></div></div><div class=package-links><a href=https://github.com/scverse/SnapATAC2 target=_blank>GitHub</a>
<a href=https://scverse.org/SnapATAC2/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/snapatac2/ target=_blank>PyPI</a>
<a href=https://anaconda.org/bioconda/snapatac2 target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/rapids-singlecell.svg alt="Logo for rapids-singlecell"></div><div class=package-info><div class=package-tile id=rapids-singlecell-tile><div class=package-text><span class=package-name>rapids-singlecell</span>
<span class=package-details>rapids-singlecell is a GPU-accelerated single-cell analysis library that serves as a drop-in replacement for scanpy, squidpy, and decoupler.</span></div></div><div class=package-links><a href=https://github.com/scverse/rapids_singlecell target=_blank>GitHub</a>
<a href=https://rapids-singlecell.readthedocs.io/en/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/rapids-singlecell/ target=_blank>PyPI</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/pertpy.svg alt="Logo for pertpy"></div><div class=package-info><div class=package-tile id=pertpy-tile><div class=package-text><span class=package-name>pertpy</span>
<span class=package-details>Pertpy is a framework for analyzing large-scale single-cell perturbation experiments. It harmonizes datasets, automates metadata annotation, calculates perturbation distances, and analyzes cellular responses to genetic modifications, drugs, and environmental changes.</span></div></div><div class=package-links><a href=https://github.com/scverse/pertpy target=_blank>GitHub</a>
<a href=https://pertpy.readthedocs.io/en/stable/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/pertpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/pertpy target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/decoupler.svg alt="Logo for decoupler"></div><div class=package-info><div class=package-tile id=decoupler-tile><div class=package-text><span class=package-name>decoupler</span>
<span class=package-details>decoupler is a framework containing different enrichment statistical methods to extract biologically driven scores from omics data within a unified framework.</span></div></div><div class=package-links><a href=https://github.com/scverse/decoupler target=_blank>GitHub</a>
<a href=https://decoupler.readthedocs.io/en/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/decoupler/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/decoupler-py target=_blank>Conda</a></div></div></div></div><h2 id=ecosystem>Ecosystem packages maintained by scverse community</h2><div><p><p>Many popular packages rely on scverse functionality. For instance, they take advantage of established data format standards such as AnnData and MuData, or are designed to be integrated into the workflow of analysis frameworks. Here, we list ecosystem packages following development best practices (continuous testing, documented, available through standard distribution tools).</p><p><em>This listing is a work in progress. See <a href=https://github.com/scverse/ecosystem-packages>scverse/ecosystem-packages</a> for inclusion criteria, and to submit more packages.</em></p></p></div><div id=ecosystem-packages><input type=text class=form-control id=eco-filter onkeyup=filterPackages() placeholder="Search through 73 packages" title="Type in your search query"><table class=table id=eco-table><thead><tr><th scope=col>Package</th><th scope=col>Description</th></tr></thead><tbody><tr class="package-links eco-table-row"><td><a href=https://github.com/quadbio/cell-annotator target=_blank>CellAnnotator</a></td><td>CellAnnotator is a leightweight tool to query large language models for cell type labels in scRNA-seq data. It can incorporate prior knowledge, and it creates consistent labels across samples in your study.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/CSOgroup/cellcharter target=_blank>CellCharter</a></td><td>CellCharter is a framework to identify, characterize and compare spatial domains from spatial omics and multi-omics data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/quadbio/cellmapper target=_blank>CellMapper</a></td><td>CellMapper is a leightweight tool to transfer labels, expression values and embeddings from reference to query datasets using k-NN mapping. It&rsquo;s fast and versatile, applicable to mapping scenarios in space, across modalities, or from an atlas to a new query dataset.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/morris-lab/CellOracle target=_blank>CellOracle</a></td><td>A computational tool that integrates single-cell transcriptome and epigenome profiles
to infer gene regulatory networks (GRNs), critical regulators of cell identity.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/cellrank target=_blank>CellRank</a></td><td>CellRank is a toolkit to uncover cellular dynamics based on Markov state modeling of single-cell data.
It contains two main modules - kernels compute cell-cell transition probabilities and estimators generate
hypothesis based on these.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/gao-lab/Cell_BLAST target=_blank>Cell_BLAST</a></td><td>Cell BLAST is a cell querying tool for single-cell transcriptomics data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/ventolab/CellphoneDB target=_blank>CellphoneDB</a></td><td>CellphoneDB is a publicly available repository of HUMAN curated receptors, ligands and their interactions paired with a tool to interrogate your own single-cell transcriptomics data (or even bulk transcriptomics data if your samples represent pure populations!). A distinctive feature of CellphoneDB is that the subunit architecture of either ligands and receptors is taken into account, representing heteromeric complexes accurately. This is crucial, as cell communication relies on multi-subunit protein complexes that go beyond the binary representation used in most databases and studies. CellphoneDB also incorporates biosynthetic pathways in which we use the last representative enzyme as a proxy of ligand abundance, by doing so, we include interactions involving non-peptidic molecules. CellphoneDB includes only manually curated and reviewed molecular interactions with evidenced role in cellular communication.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lilab-bcb/cirrocumulus target=_blank>Cirrocumulus</a></td><td>Cirrocumulus is an interactive visualization tool for large-scale single-cell genomics data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/JonathanShor/DoubletDetection target=_blank>DoubletDetection</a></td><td>DoubletDetection is a Python3 package to detect doublets (technical errors) in single-cell
RNA-seq count matrices.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/huang-sh/GPTBioInsightor target=_blank>GPTBioInsightor</a></td><td>GPTBioInsightor is a tool designed for single-cell data analysis, particularly beneficial for newcomers to a biological field or those in interdisciplinary areas who may lack sufficient biological background knowledge.
GPTBioInsightor utilizes the powerful capabilities of large language models to help people quickly gain knowledge and insight, enhancing their work efficiency.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/cantinilab/GRnnData target=_blank>GRnnData</a></td><td>An onverload of anndata to more easily work with gene networks.
Allows easy conversion between anndata and grnndata and provide loads of usefull utilities functions.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/rendeirolab/lazyslide target=_blank>LazySlide</a></td><td>LazySlide is a Python library for processing whole slide images (WSI) analysis. It provides a simple interface to perform robust preprocessing and advanced analysis for WSI.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/cantinilab/mowgli target=_blank>Mowgli</a></td><td>Paired single-cell multi-omics data integration with Optimal Transport-flavored Nonnegative Matrix Factorization</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/welch-lab/MultiVelo target=_blank>Multivelo</a></td><td>A mechanistic model of gene expression that extends the popular RNA velocity
framework by incorporating epigenomic data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/loosolab/PEAKQC target=_blank>PEAKQC</a></td><td>periodicity evaluation in scATAC-seq data for quality assessment</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/CostaLab/PILOT target=_blank>PILOT</a></td><td>PILOT is a Python library for Detection of PatIent-Level distances from single cell genomics and pathomics data with Optimal Transport.</td></tr><tr class="package-links eco-table-row"><td><a href=https://pathml.org/ target=_blank>PathML</a></td><td>An open-source toolkit for computational pathology and machine learning.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/owkin/PyDESeq2 target=_blank>PyDESeq2</a></td><td>PyDESeq2 is a python package for bulk RNA-seq differential expression analysis. It is a re-implementation
from scratch of the main features of the R package DESeq2 (Love et al. 2014).</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/ComputationalBiomedicineGroup/Rectangle target=_blank>Rectangle</a></td><td>Rectangle is a python package for computational deconvolution.
Rectangle presents a novel approach to second-generation deconvolution, characterized by hierarchical processing, an estimation of unknown cellular content and a significant reduction in data volume during signature matrix computation.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/linbuliao/SC2Spa target=_blank>SC2Spa</a></td><td>SC2Spa is a deep learning-based tool for predicting the spatial coordinates of single cells based on transcriptome.
Two paired single cell and spatial transcriptomic datasets are required to run SC2Spa. SC2Spa is trained on a ST reference
dataset to learn the relationship of gene expression and spatial coordinates. The trained fully-connected neural network
can be used to predict the locations of a single cell with only the transcriptomic profile as input. The predicted locations
of single cells can be further used to study the communication of the single cells.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/jsxlei/SCALEX target=_blank>SCALEX</a></td><td>SCALEX is an integration and projection tool for atlas-level single-cell RNA-seq and ATAC-seq data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/dynverse/anndata target=_blank>anndata for R</a></td><td>A &lsquo;reticulate&rsquo; wrapper for the Python package &lsquo;anndata&rsquo;. Provides a scalable way of keeping track of data and learned annotations. Used to read from and write to the h5ad file format.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/srivarra/annsel target=_blank>annsel</a></td><td>Annsel is a user-friendly library that brings familiar dataframe-style operations to AnnData objects such as
selection, filtering and group by&rsquo;s.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/jkobject/benGRN target=_blank>benGRN</a></td><td>Benchmarking tool for gene network inference from single cell RNAseq methods.
It uses the grnndata/anndata modality and only contains biological ground truth networks</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/ckmah/bento-tools target=_blank>bento-tools</a></td><td>A Python toolkit for subcellular analysis of spatial transcriptomics data</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/nitzanlab/biolord target=_blank>biolord</a></td><td>biolord (biological representation disentanglement) is a deep generative framework for disentangling known and unknown attributes in single-cell data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/BayraktarLab/cell2location target=_blank>cell2location</a></td><td>Cell2location is a Bayesian model that can resolve fine-grained cell types in spatial transcriptomic data
and create comprehensive cellular maps of diverse tissues. Cell2location accounts for technical sources
of variation and borrows statistical strength across locations, thereby enabling the integration of single-cell
and spatial transcriptomics with higher sensitivity and resolution than existing tools.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/chanzuckerberg/cellxgene target=_blank>cellxgene</a></td><td>CZ CELLxGENE Annotate (pronounced &ldquo;cell-by-gene&rdquo;) is an interactive data explorer for single-cell datasets,
such as those coming from the Human Cell Atlas.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/zktuong/dandelion target=_blank>dandelion</a></td><td>dandelion - A single cell BCR/TCR V(D)J-seq analysis package for 10X Chromium 5&rsquo; data.
It streamlines the pre-processing, leveraging some tools from immcantation suite, and
integrates with scanpy/anndata for single-cell BCR/TCR analysis. It also includes a
couple of functions for visualization.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/aristoteleo/dynamo-release target=_blank>dynamo-release</a></td><td>Inclusive model of expression dynamics with metabolic labeling based scRNA-seq / multiomics,
vector field reconstruction, potential landscape mapping, differential geometry analyses,
and most probably paths / in silico perturbation predictions.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/colomemaria/epiScanpy target=_blank>epiScanpy</a></td><td>EpiScanpy is a toolkit to analyse single-cell open chromatin (scATAC-seq) and single-cell
DNA methylation (for example scBS-seq) data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/zunderlab/eschr target=_blank>eschr</a></td><td>ESCHR is an ensemble clustering method that provides hard clustering along with
uncertainty scores and soft clustering outputs for enhanced interpretability.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/mikelkou/fava target=_blank>fava</a></td><td>FAVA uses Variational Autoencoders to infer functional associations from
large-scale scRNA-seq (and proteomics) data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/saeyslab/FlowSOM_Python target=_blank>flowsom</a></td><td>The complete FlowSOM package known from R, now available in Python!
Analyze high-dimensional cytometry data using FlowSOM,
a clustering and visualization algorithm based on a self-organizing map (SOM).
FlowSOM is used to distinguish cell populations from cytometry data in an unsupervised way and
can help to gain deeper insights in fields such as immunology and oncology.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/IlyaLab/gssnng target=_blank>gssnng</a></td><td>Single-cell gene set scoring with nearest neighbor graph smoothed data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/yoseflab/hotspot target=_blank>hotspot</a></td><td>Hotspot is a tool for identifying informative genes (and gene modules) in a single-cell dataset.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/icbi-lab/infercnvpy target=_blank>infercnvpy</a></td><td>Infer copy number variation (CNV) from scRNA-seq data. Plays nicely with Scanpy.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/saezlab/liana-py target=_blank>liana</a></td><td>Python package to infer cell-cell communication events from omics data using a collection of methods.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/dcjones/maxspin target=_blank>maxspin</a></td><td>An information theoretic approach to detecting spatially varying genes</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/moscot target=_blank>moscot</a></td><td>moscot is a scalable toolbox for multiomics single-cell optimal transport applications.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/MICS-Lab/novae target=_blank>novae</a></td><td>Graph-based foundation model for spatial transcriptomics data.
Zero-shot spatial domain inference, batch-effect correction, and many other features.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/Starlitnightly/omicverse target=_blank>omicverse</a></td><td>OmicVerse is the fundamental package for multi omics included bulk and single cell analysis with Python.
The original name of the omicverse was Pyomic, but we wanted to address a whole universe of transcriptomics,
so we changed the name to OmicVerse, it aimed to solve all task in RNA-seq.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/dpeerlab/Palantir target=_blank>Palantir</a></td><td>Palantir is an algorithm to align cells along differentiation trajectories.
Palantir models differentiation as a stochastic process where stem cells
differentiate to terminally differentiated cells by a series of steps through
a low dimensional phenotypic manifold. Palantir effectively captures the
continuity in cell states and the stochasticity in cell fate determination.
Palantir has been designed to work with multidimensional single cell data from
diverse technologies such as Mass cytometry and single cell RNA-seq.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/DendrouLab/panpipes target=_blank>Panpipes</a></td><td>A pipeline for multiomic single-cell and spatial transcriptomic data analysis</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lilab-bcb/pegasus target=_blank>pegasus</a></td><td>Pegasus is a tool for analyzing transcriptomes of millions of single cells.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/YosefLab/popV target=_blank>popV</a></td><td>p(opular)V(oting) is a consensus tool for transfering labels from an annotated reference dataset to an unannotated query dataset. Consensus calling allows interpretable scores that quantify certainty.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/const-ae/pyLemur target=_blank>pyLemur</a></td><td>Python implementation of the LEMUR algorithm for analyzing multi-condition single-cell RNA-seq data..</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/aertslab/pySCENIC target=_blank>pySCENIC</a></td><td>pySCENIC is a lightning-fast python implementation of the SCENIC pipeline
(Single-Cell rEgulatory Network Inference and Clustering) which enables
biologists to infer transcription factors, gene regulatory networks and
cell types from single-cell RNA-seq data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lzj1769/pychromVAR target=_blank>pychromVAR</a></td><td>A python pacakge for chromVAR.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/complextissue/pytximport target=_blank>pytximport</a></td><td>A Python port of the <code>tximport</code> R package for importing transcript-level quantification data from various RNA-seq quantification tools such as <code>salmon</code> and <code>kallisto</code> and summarizing it to the gene level.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/earmingol/scCellFie target=_blank>scCellFie</a></td><td>scCellFie infers metabolic activities from single-cell and spatial transcriptomics and offers a variety of downstream analyses.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/jkobject/scDataLoader target=_blank>scDataLoader</a></td><td>A dataloader for large single cell databases like cellxgene.
Does weighted random sampling, downloading and preprocessing.
works with anndata, zarr, and h5ad files.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/LouisFaure/scFates target=_blank>scFates</a></td><td>A scalable python package for tree inference and advanced pseudotime analysis from scRNAseq data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scgen target=_blank>scGen</a></td><td>scGen is a generative model to predict single-cell perturbation response
across cell types, studies and species.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/kharchenkolab/scLiTr target=_blank>scLiTr</a></td><td>scLiTr (single-cell Lineage Tracing) is a python package for exploratory analysis of
barcoding-based scRNA-Seq lineage tracing experiments</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/cantinilab/scPRINT target=_blank>scPRINT</a></td><td>A single cell foundation model for Gene network inference and more&mldr;</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/loosolab/scanpro target=_blank>scanpro</a></td><td>robust cell proportion analysis for single cell data</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/dawe/schist target=_blank>schist</a></td><td>schist applies Stochastic Block Models (SBM) to the analysis of single cell data, in particular to identify cell populations</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scib target=_blank>scib</a></td><td>Evaluating single-cell data integration methods</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/scmcphub target=_blank>scmcp</a></td><td>A MCP server hub for scRNA-Seq analysis software.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/frankligy/scTriangulate target=_blank>scTriangulate</a></td><td>Python package to mix-and-match conflicting clustering results in single cell analysis and generate reconciled clustering solutions</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scvelo target=_blank>scVelo</a></td><td>scVelo is a scalable toolkit for RNA velocity analysis in single cells, based on <a href=https://doi.org/10.1038/s41587-020-0591-3>Bergen et al., Nature Biotech, 2020</a>.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/MICS-Lab/scyan target=_blank>scyan</a></td><td>Biology-driven deep generative model for cell-type annotation in cytometry.
Scyan is an interpretable model that also corrects batch-effect and
can be used for debarcoding or population discovery.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/sfaira target=_blank>sfaira</a></td><td>sfaira is a model and a data repository in a single python package.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/nitzanlab/sift-sc target=_blank>sift-sc</a></td><td>SiFT is a computational framework which aims to uncover the underlying structure by filtering out previously exposed biological signals.
SiFT can be applied to a wide range of tasks, from (i) the removal of unwanted variation as a pre-processing step, through (ii) revealing hidden biological structure by utilizing prior knowledge with respect to existing signal, to (iii) uncovering trajectories of interest using reference data to remove unwanted variation.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/NKI-CCB/sobolev_alignment target=_blank>Sobolev Alignment</a></td><td>Sobolev alignment of deep probabilistic models for comparing single cell profiles from pre-clinical models and patients</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/gustaveroussy/sopa target=_blank>sopa</a></td><td>Technology-invariant pipeline for spatial-omics analysis that
scales to millions of cells. It includes segmentation, annotation,
spatial statistics, and efficient visualization.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/almaan/eggplant target=_blank>spatial-eggplant</a></td><td>Python package designed to transfer information from multiple spatial-transcriptomics data sets to a single reference representing a Common Coordinate Framework (CCF).</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/sagar87/spatialproteomics target=_blank>spatialproteomics</a></td><td>Spatialproteomics is an interoperable toolbox for analyzing highly multiplexed fluorescence image data.
This analysis involves a sequence of steps, including segmentation, image processing,
marker quantification, cell type classification, and neighborhood analysis.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/potulabe/symphonypy target=_blank>Symphonypy</a></td><td>Symphonypy is a pure Python port of Symphony label transfer algorithm for reference-based cell type
annotation.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/broadinstitute/Tangram target=_blank>tangram</a></td><td>Spatial alignment and gene expression mapping of single cell transcriptomic data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/vitessce/vitessce target=_blank>vitessce</a></td><td>Vitessce is an integrative visualization framework for multimodal and 2D/3D spatially resolved single-cell data. It consists of reusable, interactive linked views including scatterplot embedding views,
2D/3D spatial image views, genome browser tracks, statistical plots, and control views, built on web technologies such as WebGL and WebXR.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/rendeirolab/wsidata target=_blank>wsidata</a></td><td>wsidata is a data-structure for efficient IO of Whole Slide Images based on spatialdata.</td></tr></tbody></table></div></div></article></div></div></div><footer><div id=footer-content class=container><div class=row><div class="col-12 col-md-6 col-lg-3"><h5>Pages</h5><ul class=list-unstyled><li><a href=/packages>Packages</a></li><li><a href=/learn>Learn</a></li><li><a href=/people>People</a></li><li><a href=/blog>Blog</a></li><li><a href=/events>Events</a></li><li><a href=/about>About</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><h5>Governance</h5><ul class=list-unstyled><li><a href=/about/mission/>Mission statement</a></li><li><a href=/about/code_of_conduct>Code of conduct</a></li><li><a href=/about/roles>Roles</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><h5>Join scverse</h5><ul class=list-unstyled><li><a href=https://github.com/scverse target=_blank>GitHub</a></li><li><a href=https://discourse.scverse.org/ target=_blank>Discourse</a></li><li><a href=https://scverse.zulipchat.com/ target=_blank>Zulip</a></li><li><a href=https://x.com/scverse_team target=_blank>X (Twitter)</a></li><li><a href=https://bsky.app/profile/scverse.bsky.social target=_blank>Bluesky</a></li><li><a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA target=_blank>YouTube</a></li></ul></div></div><div class=signature><p>scverse® core team, 2025</p></div></div></footer></body></html>