<!doctype html><html><head><title>Packages</title><link rel=stylesheet type=text/css href=/bootstrap/css/bootstrap.min.css><link rel=stylesheet type=text/css href=/bootstrap/icons/bootstrap-icons.css><link rel=stylesheet type=text/css href=https://scverse.org/main.min.css><link rel=apple-touch-icon sizes=180x180 href=../img/favicon/apple-touch-icon.png><link rel=icon type=image/png sizes=32x32 href=../img/favicon/favicon-32x32.png><link rel=icon type=image/png sizes=16x16 href=../img/favicon/favicon-16x16.png><link rel=manifest href=../img/favicon/site.webmanifest><link rel=mask-icon href=../img/favicon/safari-pinned-tab.svg color=#000000><link rel="shortcut icon" href=../img/favicon/favicon.ico><meta name=msapplication-TileColor content="#00aba9"><meta name=msapplication-config content="../img/favicon/browserconfig.xml"><meta name=theme-color content="#ffffff"><meta charset=utf-8><meta content="utf-8" http-equiv=encoding><meta name=twitter:card content="summary"><meta name=twitter:title content="Packages"><meta name=twitter:description content><meta property="og:title" content="Packages"><meta property="og:description" content><meta property="og:type" content="website"><meta property="og:url" content="https://scverse.org/packages/"><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><script src=/main.js async></script>
<script src=/bootstrap/js/bootstrap.bundle.min.js async></script>
<script data-goatcounter=https://scverse.goatcounter.com/count async src=//gc.zgo.at/count.js></script></head><body><header><div class=container style=max-width:850px><nav class="navbar navbar-expand-lg navbar-light"><div class=container-fluid><a class="navbar-brand row align-items-center m-0" href=/><div id=scverse-logo class=col></div><span class="logo-name col m-0 px-3">scverse</span></a>
<button class=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#navbarSupportedContent aria-controls=navbarSupportedContent aria-expanded=false aria-label="Toggle navigation">
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse justify-content-end" id=navbarSupportedContent><ul class="navbar-nav mb-2 mb-lg-0 nav-buttons"><li class="nav-item dropdown"><a class="nav-link dropdown-toggle
active" href=/packages id=packages-dropdown role=button aria-expanded=false>Packages</a><ul class=dropdown-menu aria-labelledby=packages-dropdown><li><a class=dropdown-item href=/packages#core-packages>Core</a></li><li><a class=dropdown-item href=/packages/#ecosystem>Ecosystem</a></li></ul></li><li class=nav-item><a class=nav-link href=/learn>Learn</a></li><li class=nav-item><a class=nav-link href=/people>People</a></li><li class=nav-item><a class=nav-link href=/blog>Blog</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=/about id=about-dropdown role=button aria-expanded=false>About</a><ul class=dropdown-menu aria-labelledby=about-dropdown><li><a class=dropdown-item href=/about>About scverse</a></li><li><a class=dropdown-item href=/about/mission>Mission statement</a></li><li><a class=dropdown-item href=/about/roles>Roles</a></li><li><a class=dropdown-item href=/about/code_of_conduct>Code of Conduct</a></li></ul></li><li class=nav-item><a id=join-button class="nav-link nav-button-hl" href=/join>Join</a></li></ul></div></div></nav></div></header><div id=wrapper><div id=content><div id=page-content><h1>Packages</h1><article class=post><div class=post-content id=packages-content><h2 id=core-packages>Core packages</h2><div id=packages-list><div class=package-section><div class=package-img><img src=../img/libs/anndata_schema.svg></div><div class=package-info><div class=package-tile id=anndata-tile><div class=package-text><span class=package-name>anndata</span>
<span class=package-details>Anndata is a Python package for handling annotated data matrices in memory and on disk, positioned between pandas and xarray. anndata offers a broad range of computationally efficient features including, among others, sparse data support, lazy operations, and a PyTorch interface.</span></div></div><div class=package-links><a href=https://github.com/theislab/anndata target=_blank>GitHub</a>
<a href=https://anndata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/anndata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/anndata target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/libs/mudata_flat.svg></div><div class=package-info><div class=package-tile id=mudata-tile><div class=package-text><span class=package-name>mudata</span>
<span class=package-details>MuData is a format for annotated multimodal datasets where each modality is represented by an AnnData object. MuData's reference implementation is in Python, and the cross-language functionality is achieved via HDF5-based .h5mu files with libraries in R and Julia.</span></div></div><div class=package-links><a href=https://github.com/scverse/mudata target=_blank>GitHub</a>
<a href=https://mudata.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://pypi.org/project/mudata/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/mudata target=_blank>Conda</a>
<a href=https://github.com/scverse/Muon.jl target=_blank>Muon.jl</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/scanpy.svg></div><div class=package-info><div class=package-tile id=scanpy-tile><div class=package-text><span class=package-name>scanpy</span>
<span class=package-details>Scanpy is a scalable toolkit for analyzing single-cell gene expression data built jointly with anndata. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</span></div></div><div class=package-links><a href=https://github.com/theislab/scanpy target=_blank>GitHub</a>
<a href=https://scanpy.readthedocs.io/en/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scanpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/conda-forge/scanpy target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/muon.svg></div><div class=package-info><div class=package-tile id=muon-tile><div class=package-text><span class=package-name>muon</span>
<span class=package-details>muon is a Python framework for multimodal omics analysis. While there are many features that muon brings to the table, there are three key areas that its functionality is focused on.</span></div></div><div class=package-links><a href=https://github.com/scverse/muon target=_blank>GitHub</a>
<a href=https://muon.readthedocs.io/en/latest/ target=_blank>Documentation</a>
<a href=https://muon-tutorials.readthedocs.io/en/latest/ target=_blank>Tutorials</a>
<a href=https://pypi.org/project/muon/ target=_blank>PyPI</a>
<a href=https://gtca.github.io/muon/ target=_blank>Website</a></div></div></div><div class=package-section><div class=package-img><img src=../img/libs/scvi_tools_graph_model.svg></div><div class=package-info><div class=package-tile id=scvi-tools-tile><div class=package-text><span class=package-name>scvi-tools</span>
<span class=package-details>scvi-tools is a library for developing and deploying machine learning models based on PyTorch and AnnData. With an emphasis on probablistic models, scvi-tools steamlines the development process via training, data management, and user interface abstractions. scvi-tools also contains easy-to-use implementations of more than 14 state-of-the-art probabilistic models in the field.</span></div></div><div class=package-links><a href=https://github.com/yoseflab/scvi-tools target=_blank>GitHub</a>
<a href=https://docs.scvi-tools.org/en/stable/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scvi-tools/ target=_blank>PyPI</a>
<a href=https://scvi-tools.org/ target=_blank>Website</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/scirpy.svg></div><div class=package-info><div class=package-tile id=scirpy-tile><div class=package-text><span class=package-name>scirpy</span>
<span class=package-details>Scirpy is a scalable toolkit to analyse T-cell receptor or B-cell receptor repertoires from single-cell RNA sequencing data. It seamlessly integrates with scanpy and provides various modules for data import, analysis and visualization.</span></div></div><div class=package-links><a href=https://github.com/icbi-lab/scirpy target=_blank>GitHub</a>
<a href=https://icbi-lab.github.io/scirpy/latest/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/scirpy/ target=_blank>PyPI</a>
<a href=https://anaconda.org/bioconda/scirpy target=_blank>Conda</a></div></div></div><div class=package-section><div class=package-img><img src=../img/icons/squidpy.svg></div><div class=package-info><div class=package-tile id=squidpy-tile><div class=package-text><span class=package-name>squidpy</span>
<span class=package-details>Squidpy is a tool for the analysis and visualization of spatial molecular data. It builds on top of scanpy and anndata, from which it inherits modularity and scalability. It provides analysis tools that leverages the spatial coordinates of the data, as well as tissue images if available.</span></div></div><div class=package-links><a href=https://github.com/theislab/squidpy target=_blank>GitHub</a>
<a href=https://squidpy.readthedocs.io/ target=_blank>Documentation and tutorials</a>
<a href=https://pypi.org/project/squidpy/ target=_blank>PyPI</a></div></div></div></div><h2 id=ecosystem>Ecosystem</h2><div><p><p>Many popular packages rely on scverse functionality. For instance, they take advantage of established data format standards such as AnnData and MuData, or are designed to be integrated into the workflow of analysis frameworks. Here, we list ecosystem packages following development best practices (continuous testing, documented, available through standard distribution tools).</p><p><em>This listing is a work in progress. See <a href=https://github.com/scverse/ecosystem-packages>scverse/ecosystem-packages</a> for inclusion criteria, and to submit more packages.</em></p></p></div><div id=ecosystem-packages><input type=text class=form-control id=eco-filter onkeyup=filterPackages() placeholder="Search through 33 packages" title="Type in your search query"><table class=table id=eco-table><thead><tr><th scope=col>Package</th><th scope=col>Description</th></tr></thead><tbody><tr class="package-links eco-table-row"><td><a href=https://github.com/morris-lab/CellOracle target=_blank>CellOracle</a></td><td>A computational tool that integrates single-cell transcriptome and epigenome profiles
to infer gene regulatory networks (GRNs), critical regulators of cell identity.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/cellrank target=_blank>CellRank</a></td><td>CellRank is a toolkit to uncover cellular dynamics based on Markov state modeling of single-cell data.
It contains two main modules - kernels compute cell-cell transition probabilities and estimators generate
hypothesis based on these.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/gao-lab/Cell_BLAST target=_blank>Cell_BLAST</a></td><td>Cell BLAST is a cell querying tool for single-cell transcriptomics data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lilab-bcb/cirrocumulus target=_blank>Cirrocumulus</a></td><td>Cirrocumulus is an interactive visualization tool for large-scale single-cell genomics data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/JonathanShor/DoubletDetection target=_blank>DoubletDetection</a></td><td>DoubletDetection is a Python3 package to detect doublets (technical errors) in single-cell
RNA-seq count matrices.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/cantinilab/mowgli target=_blank>Mowgli</a></td><td>Paired single-cell multi-omics data integration with Optimal Transport-flavored Nonnegative Matrix Factorization</td></tr><tr class="package-links eco-table-row"><td><a href=https://pathml.org/ target=_blank>PathML</a></td><td>An open-source toolkit for computational pathology and machine learning.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/owkin/PyDESeq2 target=_blank>PyDESeq2</a></td><td>PyDESeq2 is a python package for bulk RNA-seq differential expression analysis. It is a re-implementation
from scratch of the main features of the R package DESeq2 (Love et al. 2014).</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/kaizhang/SnapATAC2 target=_blank>SnapATAC2</a></td><td><p>SnapATAC2 is the successor of the SnapATAC R package, featuring:</p><ul><li>Faster and less memory usage, scale to >1M cells.</li><li>Improved dimension reduction and sampling algorithm.</li></ul></td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/dynverse/anndata target=_blank>anndata for R</a></td><td>A &lsquo;reticulate&rsquo; wrapper for the Python package &lsquo;anndata&rsquo;. Provides a scalable way of keeping track of data and learned annotations. Used to read from and write to the h5ad file format.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/ckmah/bento-tools target=_blank>bento-tools</a></td><td>A Python toolkit for subcellular analysis of spatial transcriptomics data</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/nitzanlab/biolord target=_blank>biolord</a></td><td>biolord (biological representation disentanglement) is a deep generative framework for disentangling known and unknown attributes in single-cell data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/BayraktarLab/cell2location target=_blank>cell2location</a></td><td>Cell2location is a Bayesian model that can resolve fine-grained cell types in spatial transcriptomic data
and create comprehensive cellular maps of diverse tissues. Cell2location accounts for technical sources
of variation and borrows statistical strength across locations, thereby enabling the integration of single-cell
and spatial transcriptomics with higher sensitivity and resolution than existing tools.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/chanzuckerberg/cellxgene target=_blank>cellxgene</a></td><td>CZ CELLxGENE Annotate (pronounced &ldquo;cell-by-gene&rdquo;) is an interactive data explorer for single-cell datasets,
such as those coming from the Human Cell Atlas.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/zktuong/dandelion target=_blank>dandelion</a></td><td>dandelion - A single cell BCR/TCR V(D)J-seq analysis package for 10X Chromium 5&rsquo; data.
It streamlines the pre-processing, leveraging some tools from immcantation suite, and
integrates with scanpy/anndata for single-cell BCR/TCR analysis. It also includes a
couple of functions for visualization.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/saezlab/decoupler-py target=_blank>decoupler</a></td><td>Python package to infer biological activities from omics data using a collection of methods.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/aristoteleo/dynamo-release target=_blank>dynamo-release</a></td><td>Inclusive model of expression dynamics with metabolic labeling based scRNA-seq / multiomics,
vector field reconstruction, potential landscape mapping, differential geometry analyses,
and most probably paths / in silico perturbation predictions.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/colomemaria/epiScanpy target=_blank>epiScanpy</a></td><td>EpiScanpy is a toolkit to analyse single-cell open chromatin (scATAC-seq) and single-cell
DNA methylation (for example scBS-seq) data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/yoseflab/hotspot target=_blank>hotspot</a></td><td>Hotspot is a tool for identifying informative genes (and gene modules) in a single-cell dataset.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/icbi-lab/infercnvpy target=_blank>infercnvpy</a></td><td>Infer copy number variation (CNV) from scRNA-seq data. Plays nicely with Scanpy.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lilab-bcb/pegasus target=_blank>pegasus</a></td><td>Pegasus is a tool for analyzing transcriptomes of millions of single cells.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/aertslab/pySCENIC target=_blank>pySCENIC</a></td><td>pySCENIC is a lightning-fast python implementation of the SCENIC pipeline
(Single-Cell rEgulatory Network Inference and Clustering) which enables
biologists to infer transcription factors, gene regulatory networks and
cell types from single-cell RNA-seq data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/lzj1769/pychromVAR target=_blank>pychromVAR</a></td><td>A python pacakge for chromVAR.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/LouisFaure/scFates target=_blank>scFates</a></td><td>A scalable python package for tree inference and advanced pseudotime analysis from scRNAseq data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scgen target=_blank>scGen</a></td><td>scGen is a generative model to predict single-cell perturbation response
across cell types, studies and species.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scib target=_blank>scib</a></td><td>Evaluating single-cell data integration methods</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/frankligy/scTriangulate target=_blank>scTriangulate</a></td><td>Python package to mix-and-match conflicting clustering results in single cell analysis and generate reconciled clustering solutions</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/scvelo target=_blank>scVelo</a></td><td>scVelo is a scalable toolkit for RNA velocity analysis in single cells, based on <a href=https://doi.org/10.1038/s41587-020-0591-3>Bergen et al., Nature Biotech, 2020</a>.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/theislab/sfaira target=_blank>sfaira</a></td><td>sfaira is a model and a data repository in a single python package.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/nitzanlab/sift-sc target=_blank>sift-sc</a></td><td>SiFT is a computational framework which aims to uncover the underlying structure by filtering out previously exposed biological signals.
SiFT can be applied to a wide range of tasks, from (i) the removal of unwanted variation as a pre-processing step, through (ii) revealing hidden biological structure by utilizing prior knowledge with respect to existing signal, to (iii) uncovering trajectories of interest using reference data to remove unwanted variation.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/almaan/eggplant target=_blank>spatial-eggplant</a></td><td>Python package designed to transfer information from multiple spatial-transcriptomics data sets to a single reference representing a Common Coordinate Framework (CCF).</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/broadinstitute/Tangram target=_blank>tangram</a></td><td>Spatial alignment and gene expression mapping of single cell transcriptomic data.</td></tr><tr class="package-links eco-table-row"><td><a href=https://github.com/vitessce/vitessce target=_blank>vitessce</a></td><td>Vitessce consists of reusable interactive views including a scatterplot,
spatial+imaging plot, genome browser tracks, statistical plots, and control
views, built on web technologies such as WebGL.</td></tr></tbody></table></div></div></article></div></div></div><footer><div id=footer-content class=container><div class=row><div class="col-12 col-md-6 col-lg-3"><ul><h5>Pages</h5><li><a href=/packages>Packages</a></li><li><a href=/learn>Learn</a></li><li><a href=/people>People</a></li><li><a href=/blog>Blog</a></li><li><a href=/about>About</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Governance</h5><li><a href=/about/mission/>Mission statement</a></li><li><a href=/about/code_of_conduct>Code of conduct</a></li><li><a href=/about/roles>Roles</a></li></ul></div><div class="col-12 col-md-6 col-lg-3"><ul><h5>Join scverse</h5><li><a href=https://github.com/scverse target=_blank>GitHub</a></li><li><a href=https://discourse.scverse.org/ target=_blank>Discourse</a></li><li><a href=https://scverse.zulipchat.com/ target=_blank>Zulip</a></li><li><a href=https://twitter.com/scverse_team target=_blank>Twitter</a></li><li><a href=https://www.youtube.com/channel/UCpsvsIAW3R5OdftJKKuLNMA target=_blank>YouTube</a></li></ul></div></div><div class=signature><p>scverse core team, 2022</p></div></div></footer></body></html>